### PR TITLE
[feat]: [CDS-78967]: Changes for incorporating the YAML changes in CRUD and Executing the V1 pipeline as per new YAML structure

### PR DESCRIPTION
**Pull request by brijeshdhakar**
## Describe your changes

  Changes for incorporating the YAML changes in CRUD and Executing the V1 pipeline as per new YAML structure

Changes: Taking the pipeline name and identifier from the metadata and not from the YAML. And making pipeline identifier as an mandatory field for create-pipeline flow for V1 pipelines.
And Did some refactoring to pass the pipeline-identifier from the resource layer.

Testing:
CRUD
- Tested the V0 pipeline flow for Creating a pipeline. 
- Tested the V0 pipeline flow for cloning a pipeline.
- Tested the V1 pipeline flow for Creating a pipeline. 
- Tested the V1 pipeline flow for cloning a pipeline.

Execution
- Tested executing the V1 pipeline with inputs and having V0 stages and steps inside it.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)

### CODE REVIEW

**Comment by brijeshdhakar** [comment]
trigger uts

**Comment by brijeshdhakar** [comment]
trigger uts

**Review by utkarshchoubey-harness — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, 1 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 5 file(s) to review. 0 of 2 approvals obtained.

**Review by utkarshchoubey-harness — DISMISSED** [review]

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java [hunk 1]
```diff
@@ -128,9 +128,9 @@ private String getYamlVersionFromStageField(YamlField currField) {
   }
   @Override
   public GraphLayoutResponse getLayoutNodeInfo(PlanCreationContext ctx, YamlField config) {
-    // Create graphLayout only if stages node is child of parent.(Return empty if its child of parallel.spec)
-    if (EmptyPredicate.isEmpty(config.getNode().getParentNode().getType())
-        || !config.getNode().getParentNode().getType().equals(YAMLFieldNameConstants.PIPELINE)) {
+    // Create graphLayout only if stages node is grand child of pipeline.(Return empty if its child of parallel.spec)
+    if (EmptyPredicate.isEmpty(config.getNode().getParentNode().getParentNode().getType())
```
Comment (shalini-agr) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java:132]: we decided not to get any information from parentNode, right?
Can't we add this type value also in dependency?

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 4 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 132 at r2](https://reviewable.io/reviews/harness/harness-core/52626#-Ne80VMe4I7ymr7qvh-G:-Ne80VMe4I7ymr7qvh-H:b-a44tym) ([raw file](https://github.com/harness/harness-core/blob/c471c4e88b6bfa9c86a53a85db894daf0f6e56e8/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L132)):*
> ```Java
>   public GraphLayoutResponse getLayoutNodeInfo(PlanCreationContext ctx, YamlField config) {
>     // Create graphLayout only if stages node is grand child of pipeline.(Return empty if its child of parallel.spec)
>     if (EmptyPredicate.isEmpty(config.getNode().getParentNode().getParentNode().getType())
> ```

add check for null before doing another get, extract as another node rather than in same condition

**Review by brijeshdhakar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 4 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 132 at r2](https://reviewable.io/reviews/harness/harness-core/52626#-Ne80VMe4I7ymr7qvh-G:-Ne86dIm1Y-E-v00Xg-R:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c471c4e88b6bfa9c86a53a85db894daf0f6e56e8/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L132)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

add check for null before doing another get, extract as another node rather than in same condition
</blockquote></details>

Done.
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 132 at r2](https://reviewable.io/reviews/harness/harness-core/52626#gh-1322865299:-Ne86dYdAT_7R1iy24fd:bc4bym) ([raw file](https://github.com/harness/harness-core/blob/c471c4e88b6bfa9c86a53a85db894daf0f6e56e8/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L132)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

we decided not to get any information from parentNode, right?
Can't we add this type value also in dependency?
</blockquote></details>

I think this is fine as we are not traversing to other nodes(siblings to extract much info) from the parent. But just checking the parent Node type.

**Review by archit-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 4 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 132 at r2](https://reviewable.io/reviews/harness/harness-core/52626#gh-1322865299:-Ne87_Y21a7lWzyXAnc1:b32v57) ([raw file](https://github.com/harness/harness-core/blob/c471c4e88b6bfa9c86a53a85db894daf0f6e56e8/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L132)):*
<details><summary><i>Previously, brijeshdhakar (Brijesh Dhakar) wrote…</i></summary><blockquote>

I think this is fine as we are not traversing to other nodes(siblings to extract much info) from the parent. But just checking the parent Node type.
</blockquote></details>

+1

**Comment by brijeshdhakar** [comment]
trigger uts

**Comment by brijeshdhakar** [comment]
trigger uts

**Comment by brijeshdhakar** [comment]
trigger uts

**Review by utkarshchoubey-harness — APPROVED** [review]

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 3 of 4 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 4 file(s) to review. 1. unresolved discussions.

**Review by shalini-agr — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 132 at r2](https://reviewable.io/reviews/harness/harness-core/52626#gh-1322865299:-NeC7XctBpdfEauW9wt7:b-w0yspi) ([raw file](https://github.com/harness/harness-core/blob/c471c4e88b6bfa9c86a53a85db894daf0f6e56e8/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L132)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

+1
</blockquote></details>

got it.

**Review by shalini-agr — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 3 of 4 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 1. unresolved discussions.

**Comment by brijeshdhakar** [comment]
trigger review

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java [hunk 2]
```diff
@@ -164,6 +163,15 @@ public GraphLayoutResponse getLayoutNodeInfo(PlanCreationContext ctx, YamlField
         .build();
   }
 
+  private boolean isStagesNodeGrandChildOfPipelineNode(YamlField config) {
+    YamlNode parentYamlNode = config.getNode().getParentNode();
+    if (parentYamlNode != null && parentYamlNode.getParentNode() != null
+        && YAMLFieldNameConstants.PIPELINE.equals(parentYamlNode.getParentNode().getType())) {
```
Comment (shalini-agr) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java:169]: @brijeshdhakar this should now become getKind() right?

**Review by archit-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 1. unresolved discussions.

**Review by shalini-agr — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 1 change requests. 1. unresolved discussions. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 169 at r4](https://reviewable.io/reviews/harness/harness-core/52626#-NeCUZfICZ7MWGGha-Kv:-NeCUZfICZ7MWGGha-Kw:bcjc9h3) ([raw file](https://github.com/harness/harness-core/blob/6ca8dfa0a22780ad1e20a0ce01baf7c7e382e440/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L169)):*
> ```Java
>     YamlNode parentYamlNode = config.getNode().getParentNode();
>     if (parentYamlNode != null && parentYamlNode.getParentNode() != null
>         && YAMLFieldNameConstants.PIPELINE.equals(parentYamlNode.getParentNode().getType())) {
> ```

this should now become getKind() right?

**Review by brijeshdhakar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 1 change requests. 2. unresolved discussions. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 169 at r4](https://reviewable.io/reviews/harness/harness-core/52626#-NeCUZfICZ7MWGGha-Kv:-NeCUwEuEXkLfioz-eOJ:bjc4wu9) ([raw file](https://github.com/harness/harness-core/blob/6ca8dfa0a22780ad1e20a0ce01baf7c7e382e440/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L169)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

this should now become getKind() right?
</blockquote></details>

Type would be present in the YamlField because type is Injected into the YAML before starting the planCreation in ExecutionHelper. PlanCreation totally depends on type only. 
I can explain why we have to inject Type and not use kind.

```switch (version) {  
  case PipelineVersion.V1:  
    currentProcessedYaml = YamlUtils.injectUuidWithType(pipelineYaml, YAMLFieldNameConstants.PIPELINE);  
    break;  case PipelineVersion.V0:  
    currentProcessedYaml = YamlUtils.injectUuid(pipelineYaml);  
    break;  default:  
    throw new IllegalStateException("version not supported");  
}```


Anyways I am thinking of removing this check logic altogether by simplifying some logic. Will do in the next PR.
```

**Review by shalini-agr — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 1. unresolved discussions. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 169 at r4](https://reviewable.io/reviews/harness/harness-core/52626#-NeCUZfICZ7MWGGha-Kv:-NeC_Be51t0i32VCoDfX:bbtjr4l) ([raw file](https://github.com/harness/harness-core/blob/6ca8dfa0a22780ad1e20a0ce01baf7c7e382e440/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L169)):*
<details><summary><i>Previously, brijeshdhakar (Brijesh Dhakar) wrote…</i></summary><blockquote>

Type would be present in the YamlField because type is Injected into the YAML before starting the planCreation in ExecutionHelper. PlanCreation totally depends on type only. 
I can explain why we have to inject Type and not use kind.

```switch (version) {  
  case PipelineVersion.V1:  
    currentProcessedYaml = YamlUtils.injectUuidWithType(pipelineYaml, YAMLFieldNameConstants.PIPELINE);  
    break;  case PipelineVersion.V0:  
    currentProcessedYaml = YamlUtils.injectUuid(pipelineYaml);  
    break;  default:  
    throw new IllegalStateException("version not supported");  
}```


Anyways I am thinking of removing this check logic altogether by simplifying some logic. Will do in the next PR.
```

</blockquote></details>

Got it, Brijesh, As discussed we can pick this task sometime later to see if we can get rid of this type logic altogether.

**Review by hextechpal — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 3 of 4 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)* status: 2. unresolved discussions. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 132 at r2](https://reviewable.io/reviews/harness/harness-core/52626#gh-1322865299:-NeE49JMDct5uhAsWyUb:b-qqbmhi) ([raw file](https://github.com/harness/harness-core/blob/c471c4e88b6bfa9c86a53a85db894daf0f6e56e8/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L132)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

got it.
</blockquote></details>

But why we are doing this let's keep this consistent and pass the dependency from a parent. Maybe we need to think of levels concept for plan creation too
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 169 at r4](https://reviewable.io/reviews/harness/harness-core/52626#-NeCUZfICZ7MWGGha-Kv:-NeE4jUpA_kCppAr9W0A:b-ci5aio) ([raw file](https://github.com/harness/harness-core/blob/6ca8dfa0a22780ad1e20a0ce01baf7c7e382e440/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L169)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

Got it, Brijesh, As discussed we can pick this task sometime later to see if we can get rid of this type logic altogether.
</blockquote></details>

Lets discuss this once

**Comment by brijeshdhakar** [comment]
> > Reviewed 1 of 2 files at r1, 3 of 4 files at r3, 4 of 4 files at r4, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/52626)_ status: 2. unresolved discussions.
> 
> _[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 132 at r2](https://reviewable.io/reviews/harness/harness-core/52626#gh-1322865299:-NeE49JMDct5uhAsWyUb:b-qqbmhi) ([raw file](https://github.com/harness/harness-core/blob/c471c4e88b6bfa9c86a53a85db894daf0f6e56e8/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L132)):_
> 
> Previously, shalini-agr (Shalini Agrawal) wrote…
> But why we are doing this let's keep this consistent and pass the dependency from a parent. Maybe we need to think of levels concept for plan creation too
> 
> _[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java` line 169 at r4](https://reviewable.io/reviews/harness/harness-core/52626#-NeCUZfICZ7MWGGha-Kv:-NeE4jUpA_kCppAr9W0A:b-ci5aio) ([raw file](https://github.com/harness/harness-core/blob/6ca8dfa0a22780ad1e20a0ce01baf7c7e382e440/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/stages/v1/StagesPlanCreatorV1.java#L169)):_
> 
> Previously, shalini-agr (Shalini Agrawal) wrote…
> Lets discuss this once

We don't need this particular change here. There will never be the case when the condition is false. So we can remove this altogether. But yes should think of a way to pass the info like we do in ambiance levels.

**Pull request merged**